src/main/java/org/dependencytrack/persistence/ProjectQueryManager.java

```diff
@@ -473,18 +502,21 @@ public Project updateProject(Project transientProject, boolean commitIndex) {
         project.setPurl(transientProject.getPurl());
         project.setSwidTagId(transientProject.getSwidTagId());
 
-        if (project.isActive() && hasActiveChild(project) && (transientProject.isActive() == null || !transientProject.isActive()) ){
-            LOGGER.warn(project + " cannot be set to inactive, if active children are present. " + project + " remains active");
+        if (project.isActive() && !Boolean.TRUE.equals(transientProject.isActive()) && hasActiveChild(project)){
+            throw new IllegalArgumentException("Project cannot be set to inactive if active children are present.");
         } else {
             project.setActive(transientProject.isActive());
         }
 
         if (transientProject.getParent() != null && transientProject.getParent().getUuid() != null) {
+            if (project.getUuid().equals(transientProject.getParent().getUuid())){
+                throw new IllegalArgumentException("A project cannot select itself as a parent");
+            }
             Project parent = getObjectByUuid(Project.class, transientProject.getParent().getUuid());
-            if (!parent.isActive()){
-                LOGGER.warn("The new parent project cannot be an inactive project. " + project.getParent() + " remains the parent of " + project);
-            } else if (isChild(parent, transientProject.getUuid())){
-                LOGGER.warn("The new parent project cannot be a child of the current project. " + project.getParent() + " remains the parent of " + project);
+            if (!Boolean.TRUE.equals(parent.isActive())){
+                throw new IllegalArgumentException("An inactive project cannot be selected as a parent");
+            } else if (isChildOf(parent, transientProject.getUuid())){
+                throw new IllegalArgumentException("The new parent project cannot be a child of the current project.");
             } else {
                 project.setParent(parent);
             }
```

Comment:
```suggestion
            }
            project.setParent(parent);
```
Kannst dir das else sparen wenn im if/elseif eine exception fliegt